packages/nvidia_nat_opentelemetry/src/nat/plugins/opentelemetry/mixin/otlp_span_exporter_mixin.py

```
@@ -42,16 +44,29 @@ def __init__(self, endpoint, headers, **kwargs):
+            protocol: Transport protocol to use ('http' or 'grpc'). Defaults to 'http'.
         """
         # Initialize exporter before super().__init__() to ensure it's available
         # if parent class initialization potentially calls export_otel_spans()
-        self._exporter = OTLPSpanExporter(endpoint=endpoint, headers=headers)
+
+        if protocol == 'http':
+            self._exporter = OTLPSpanExporterHTTP(endpoint=endpoint, headers=headers)
+        elif protocol == 'grpc':
+            self._exporter = OTLPSpanExporterGRPC(endpoint=endpoint, headers=headers)
+        else:
+            raise ValueError(f"Invalid protocol: {protocol}")
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)